### PR TITLE
Added: `--connections` to toggel the merging of Edges between Nodes

### PR DESCRIPTION
**Pull request by xescugc**
This option allows have a middle stage between `--raw` which shows everything and without `--raw` which shows the maximum reduction.

The `--connections` (by default `true`) toggles the logic to merge the Edges between Nodes to show a more detailed graph and not that reduced.

### CODE REVIEW

File: cmd/generate.go [hunk 1]
```diff
@@ -86,4 +88,5 @@ func init() {
 	generateCmd.Flags().StringVar(&printerType, "printer", "dot", fmt.Sprintf("Type of printer to use for the output. Supported ones are: %s", strings.Join(printer.TypeStrings(), ",")))
 	generateCmd.Flags().BoolVar(&raw, "raw", false, "Raw means that will not use any specific logic from the provider, will just display the connections between elements. It's used by default if none of the Providers is known")
 	generateCmd.Flags().BoolVar(&clean, "clean", true, "Clean means that the generated graph will not have any Node that does not have a connection/edge")
+	generateCmd.Flags().BoolVar(&connections, "connections", true, "Connections means that we'll apply the logic of the Provider to remove resources that are not Nodes")
```
Comment (tormath1) [inline cmd/generate.go:91]:
```suggestion
	generateCmd.Flags().BoolVar(&connections, "connections", true, "Connections will apply logic of the provider to remove resources that are not nodes")
```

File: generate/hcl.go [hunk 2]
```diff
@@ -157,12 +157,16 @@ func FromHCL(fs afero.Fs, path string, opt Options) (*graph.Graph, error) {
...
+	if opt.Connections {
+		err = mutate(g, opt)
+		if err != nil {
+			return nil, err
+		}
 	}
 
-	cleanHangingEdges(g, opt)
+	if opt.Clean {
+		cleanHangingEdges(g, opt)
```
Comment (tormath1) [inline generate/hcl.go:168]: `cleanHangingEdges` can silently fail. It would be nice to catch the error ?

**Comment by xescugc** [comment]
> it could be nice to also have test with {Connections: false}. At least one, to show the difference :)

Yes but I'll do it in a single PR, I have a lot of test missing to add (`Raw`,`Clean` and `Connections` to mention some, but also for all the Providers and others).

If that's ok :)

File: cmd/generate.go [hunk 3]
```diff
@@ -88,5 +88,5 @@ func init() {
 	generateCmd.Flags().StringVar(&printerType, "printer", "dot", fmt.Sprintf("Type of printer to use for the output. Supported ones are: %s", strings.Join(printer.TypeStrings(), ",")))
 	generateCmd.Flags().BoolVar(&raw, "raw", false, "Raw means that will not use any specific logic from the provider, will just display the connections between elements. It's used by default if none of the Providers is known")
 	generateCmd.Flags().BoolVar(&clean, "clean", true, "Clean means that the generated graph will not have any Node that does not have a connection/edge")
-	generateCmd.Flags().BoolVar(&connections, "connections", true, "Connections means that we'll apply the logic of the Provider to remove resources that are not Nodes")
+	generateCmd.Flags().BoolVar(&connections, "connections", true, "Connections means that will apply the logic of the provider to remove resources that are not nodes")
```
Comment (tormath1) [inline cmd/generate.go:91]:
```suggestion
	generateCmd.Flags().BoolVar(&connections, "connections", true, "Connections will apply the logic of the provider to remove resources that are not nodes")
```

**Comment by xescugc** [comment]
Done, and fixed other commands too to remove the `means that`.

**Comment by tormath1** [comment]
LGTM :) you can RS !

**Pull request merged**